### PR TITLE
feat: add thor tradable asset script

### PR DESCRIPTION
**Pull request by 0xApotheosis**
## Description

Adds a script to dynamically generate the mapping from thor asset notation to CAIP-19.

`yarn generate:thor-asset-map`

We currently maintain a hardcoded object for this, but we should make this more dynamic.

We opted to make this a generation script, rather than a run-time conversion for safety, and to give us the chance to sanity check any changes/new assets.

An asset's trade status, however, _is_ still checked at runtime.

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A, quality of life.

## Risk

Small. If something went wrong with the generation we'd simply not be able to trade to or from the assets via THORSwap, or use them in THORSavers.

This should be easy to check at runtime.

## Testing

- Check all thor-tradable assets are still apperaing as expected
- Check all THORSaver assets are still appering as expected

### Engineering

☝️

Note, we lost a few unsupported Etherum assets, and gained some, so the actual generated list varies slightly from the hardcoded one.

Also note, we do not currently parse BNB or native THOR assets, and so running the generation script will show these as un-parsable (this is fine).

### Operations

☝️

## Screenshots (if applicable)

Screenshot 2023-06-07 at 7 28 48 pm

### CODE REVIEW

**Comment by 0xApotheosis** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #4678** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4678" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4678?utm_source=stack-comment).

**Review by gomesalexandre — APPROVED** [review]
Confirmed this does what this says on the box:

- Excludes BNB and internal THOR assets from the generated assets as we don't support these
- `src/lib/swapper/swappers/ThorchainSwapper/generated/generatedTradableThorAssetMap.json` looks sane
- removing the generated file and re-generating it produces no diff

Gave it a little testing at runtime as well:

- Confirmed the savers vaults are still happy (confirmed by going through the deposit/withdraw flow, without confirming but confirmed the THOR endpoints were still hit and returning us happy responses)
- Confirmed swapper pairs still look sane as far as THOR swapper availability goes, and we're actually getting a quote for these

Left a few, non-blocking comments, to be tackled at your convenience, otherwise happy to merge this 🤌

File: scripts/generateTradableThorAssetMap/generateTradableThorAssetMap.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+export const generateTradableThorAssetMap = async () => {
+  const thorService = axios.create(axiosConfig)
+  const response = await thorService.get<ThornodePoolResponse[]>(
+    'https://dev-daemon.thorchain.shapeshift.com/lcd/thorchain/pools',
+  )
+  switch (response.status) {
+    case 200:
+      const poolData = response.data
+      const assetIdPairs = poolData.map(getAssetIdPairFromPool).filter(isSome)
+      const assetIdPairsWithRune: AssetIdPair[] = [...assetIdPairs, ['THOR.RUNE', thorchainAssetId]]
```
[the remark this answers is not in the document]
Comment (0xApotheosis) [reply scripts/generateTradableThorAssetMap/generateTradableThorAssetMap.ts:37]: Actioned in https://github.com/shapeshift/web/pull/4678/commits/03fde46b59e0a30ebd7f4a2262246161bb49af0f.

File: scripts/generateTradableThorAssetMap/utils.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    case ThorchainChain.THOR:
+      return thorchainAssetId
+    case ThorchainChain.GAIA:
+      return cosmosAssetId
+    default:
+      return undefined
+  }
+}
+
+export const getSmartContractTokenStandardFromChainId = (
```
[the remark this answers is not in the document]
Comment (0xApotheosis) [reply scripts/generateTradableThorAssetMap/utils.ts]: Actioned in https://github.com/shapeshift/web/pull/4678/commits/03fde46b59e0a30ebd7f4a2262246161bb49af0f

**Review by gomesalexandre — APPROVED** [review]

**Pull request merged**